packages/ecdh/src/main.nr

```diff
@@ -0,0 +1,94 @@
+use dep::std::ec::tecurve::affine::{Curve, Point};
+// use crate::scalar_field::ScalarField;
+// use crate::Curve;
+// use crate::TECurveParameterTrait;
+// use crate::bjj::BabyJubJubParams;
+// use std::ec::consts::te::baby_jubjub;
+// use std::ec::tecurve::affine::Point as TEPoint;
+// @@@@@@ Core ECDH Implementation
+
+/// Converts a byte array to a field element.
+/// # Arguments
+/// * `bytes` - A fixed-size array of 32 bytes.
+/// * `big_endian` - A boolean indicating if the byte array is in big-endian format.
+/// # Returns
+/// A `Field` element representing the converted byte array.
+pub fn field_from_bytes(bytes: [u8; 32], big_endian: bool) -> Field {
+    let mut as_field: Field = 0;
+    let mut offset: Field = 1;
+
+    for i in 0..32 {
+        let index = if big_endian { 31 - i } else { i };
+        as_field += (bytes[index] as Field) * offset;
+        offset *= 256;
+    }
+
+    as_field
+}
+
+/// Generates a public key from a private key using the Baby JubJub curve.
+/// # Arguments
+/// * `private_key` - A field element representing the private key.
+/// * `base_point` - A point on the Baby JubJub curve used as the base point.
+/// # Returns
+/// A `Point` on the Baby JubJub curve representing the public key.
+fn generate_public_key(private_key: Field, base_point: Point) -> Point {
```

Comment:
```suggestion
fn generate_key(private_key: Field, base_point: Point) -> Point {
```

Since this is generic for both public key and shared key generation, naming should also be generic